web/src/main/java/org/springframework/security/web/csrf/CsrfFilter.java

```diff
@@ -174,17 +174,18 @@ public void setAccessDeniedHandler(AccessDeniedHandler accessDeniedHandler) {
 	 * @return
 	 */
 	private static boolean equalsConstantTime(String expected, String actual) {
-		byte[] expectedBytes = bytesUtf8(expected);
-		byte[] actualBytes = bytesUtf8(actual);
+		if (actual == null) {
```

Comment:
I think the code could catch more cases by doing:

```java
if (expected == actual) {
    return true;
}
if (expected == null || actual == null) {
    return false;
}
```

`MessageDigest` follows the above pattern when comparing the `byte` arrays.

Reply:
ok

Reply:
ok.

Reply:
ok.
This PR was accidentally closed by me. I submitted a new pr #9561. 
Also, when I was running tests with a lower version of JDK, I found that csrfwebfilter had the same NPE captured, so I made the same modification to csrfwebfilter.
Also, thank you for your guidance. This is my first time to participate in an open source project.